twenty-first/src/util_types/merkle_tree.rs

```diff
@@ -901,11 +971,11 @@ pub mod merkle_tree_test {
     #[proptest(cases = 10)]
     fn each_leaf_can_be_verified_individually(test_tree: MerkleTreeToTest) {
         let tree = test_tree.tree;
-        for (leaf_index, &leaf) in tree.leafs().iter().enumerate() {
+        for (leaf_index, &leaf) in tree.leafs().enumerate() {
             let authentication_path = tree.authentication_structure(&[leaf_index]).unwrap();
             let proof = MerkleTreeInclusionProof {
                 tree_height: tree.height(),
-                indexed_leafs: [(leaf_index, leaf)].into(),
+                indexed_leafs: [(leaf_index as MerkleTreeLeafIndex, leaf)].into(),
```

Comment:
Depends on how hard you want to use `MerkleTreeLeafIndex` or accept that it is an alias for `usize`.

```suggestion
                indexed_leafs: [(leaf_index, leaf)].into(),
```